src/inrange.jl

```diff
@@ -32,15 +29,26 @@ function inrange{T <: AbstractFloat}(tree::NNTree{T},
         end
         idxs[i] = idx_in_ball
     end
-    return do_return_inrange(idxs, points)
+    return idxs
 end
 
-do_return_inrange(idxs, ::AbstractVector) = idxs[1]
-do_return_inrange(idxs, ::AbstractMatrix) = idxs
+function inrange{V, T <: Number}(tree::NNTree{V}, point::AbstractVector{T}, radius::Number, sortres=false)
+    idxs = inrange(tree, Vector{T}[point], radius, sortres)
```

Comment:
Yeah, there are some optimizations to be made with regards to only querying for 1 point at a time. Could also be good to privde an API where everything is preallocated, like the vector for storing indices and distances.